src/wazuh_modules/wm_exec.c

```diff
@@ -206,57 +217,47 @@ DWORD WINAPI Reader(LPVOID args) {
 // Add process to pool
 
 void wm_append_handle(HANDLE hProcess) {
-    int i;
+    HANDLE * p_hProcess = NULL;
 
-    w_mutex_lock(&wm_children_mutex);
+    os_calloc(1, sizeof(HANDLE), p_hProcess);
+    *p_hProcess = hProcess;
 
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (!wm_children[i]) {
-            wm_children[i] = hProcess;
-            break;
-        }
+    if(!OSList_AddData(wm_children_list, (void *)p_hProcess)) {
+        merror("Child process handle %p could not be registered.", hProcess);
+        os_free(p_hProcess);
     }
-
-    w_mutex_unlock(&wm_children_mutex);
-
-    if (i == WM_POOL_SIZE)
-        merror("Child process pool is full. Couldn't register handle %p.", hProcess);
 }
 
 // Remove process from pool
 
 void wm_remove_handle(HANDLE hProcess) {
-    int i;
-
-    w_mutex_lock(&wm_children_mutex);
-
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (wm_children[i] == hProcess) {
-            wm_children[i] = 0;
-            break;
+    OSListNode * node_it = NULL;
+    HANDLE * p_hProcess = NULL;
+
+    OSList_foreach(node_it, wm_children_list) {
+        p_hProcess = (HANDLE *)node_it->data;
+        if (p_hProcess && *p_hProcess == hProcess) {
+            OSList_DeleteThisNode(wm_children_list, node_it);
+            os_free(p_hProcess);
+            return;
         }
     }
 
-    if (i == WM_POOL_SIZE)
-        merror("Child process %p not found.", hProcess);
-
-    w_mutex_unlock(&wm_children_mutex);
+    merror("Child process %p not found.", hProcess);
 }
 
 // Terminate every child process group. Doesn't wait for them!
 
 void wm_kill_children() {
     // This function may be called from a signal handler
 
-    int i;
-
-    w_mutex_lock(&wm_children_mutex);
+    HANDLE * p_hProcess = NULL;
+    OSListNode * node_it = NULL;
 
-    for (i = 0; i < WM_POOL_SIZE && wm_children[i]; i++)  {
-        TerminateProcess(wm_children[i], 127);
+    OSList_foreach(node_it, wm_children_list) {
+        p_hProcess = (HANDLE *)node_it->data;
```

Comment:
can we add a simple check  if (node_it->data) before?

Reply:
will do